doc/developer-guides/hld/hld-devicemodel.rst

```diff
@@ -502,33 +504,33 @@ Device Emulation
 ****************
 
 The DM emulates different kinds of devices, such as RTC,
-LPC, UART, PCI devices, virtio block device, etc. It is important
-for device emulation can handle I/O requests
+LPC, UART, PCI devices, and virtio block device. It is important
+that device emulation can handle I/O requests
 from different devices including PIO, MMIO, and PCI CFG
 SPACE access. For example, a CMOS RTC device may access 0x70/0x71 PIO to
 get CMOS time, a GPU PCI device may access its MMIO or PIO bar space to
 complete its framebuffer rendering, or the bootloader may access a PCI
 device's CFG SPACE for BAR reprogramming.
 
 The DM needs to inject interrupts/MSIs to its frontend devices whenever
-necessary. For example, an RTC device needs get its ALARM interrupt, or a
+necessary. For example, an RTC device needs to get its ALARM interrupt, or a
 PCI device with MSI capability needs to get its MSI.
 
-DM also provides a PIRQ routing mechanism for platform devices.
+The DM also provides a PIRQ routing mechanism for platform devices.
 
 PIO/MMIO/CFG SPACE Handler
 ==========================
 
-This chapter will do a quick introduction of different I/O requests.
+This chapter provides a quick introduction of different I/O requests.
 
 PIO Handler Register
 --------------------
 
-A PIO range structure in DM is like below, it's the parameter needed to
-register PIO handler for special PIO range:
+A PIO range structure in the DM is shown below. It's the parameter needed to
+register a PIO handler for a special PIO range:
 
-.. note:: this should be references to API documentation in
-   devicemodel/include/inout.h
+.. note:: This should be references to API documentation in
+   ``devicemodel/include/inout.h``.
```

Comment:
Is this an internal note, saying to reference the API doc?

Reply:
Yes, we shouldn't write API documentation like this, but should be extracting it from the doxygen comments.  Alas, there aren't any doxygen comments in inout.h as there should be.  (This is a note I wrote over three years ago that never got resolved.)

Reply:
I created an issue.